src/events/eventActions.js

```diff
@@ -77,10 +67,9 @@ export const startEventPolling = (queueId: number, eventId: number) => async (
         // are explicitly not the place for side effects (see
         // https://redux.js.org/faq/actions).
         dispatch(doEventActionSideEffects(action));
+        dispatch(action);
       });
 
-      dispatchOrBatch(dispatch, actions);
```

Comment:
This is not a request to change something, just an FYI -- in the Zulip webapp, we process the vast majority of events individually, but have this (very early/old) code to batch certain classes of events differently.  I think the goal is to avoid unnecessary rerenders -- e.g. if you're getting 10 new messages, avoiding adding each one individually 10 times, which might be fine in the webapp but certainly isn't something we've explicitly optimized there.

That behavior ends up forcing the message-related events (message edit, etc.) to also be batched, etc., which means it probably has bugs (e.g. from reading the below, I think batched up `reaction` events may get dropped from being processed before the message that they're reacting to; apparently we did this carefully when adding `submessage` but not `reaction`):

```
    const dispatch_event = function dispatch_event(event) {                                          
        switch (event.type) {                                                                        
            case "message": {                                                                        
                const msg = event.message;                                                           
                msg.flags = event.flags;                                                             
                if (event.local_message_id) {                                                        
                    msg.local_id = event.local_message_id;                                           
                    sent_messages.report_event_received(event.local_message_id);                     
                }                                                                                    
                messages.push(msg);                                                                  
                break;                                                                               
            }                                                                                        
                                                                                                     
            case "update_message":                                                                   
                update_message_events.push(event);                                                   
                break;                                                                               
                                                                                                     
            case "delete_message":                                                                   
            case "submessage":                                                                       
            case "update_message_flags":                                                             
                post_message_events.push(event);                                                     
                break;                                                                               
                                                                                                     
            default:                                                                                 
                server_events_dispatch.dispatch_normal_event(event);                                 
        }                                                                                            
    };                                                                                               

```

For non-message events where we're confident it'd be correct to do batching, once we have long-term event queues, probably the server will eventually do whatever batching transformations are safe/correct just to safe memory.

So mostly this is agreement with a "You probably don't want to do batching unless you really have to".